### PR TITLE
Improve README steps to build the system [skip ci].

### PR DESCRIPTION
**Pull request by coryan**
Reduce the number of dependencies to install.

Add documentation for OpenSuSE, clarify that the
steps for Ubuntu Bionic also work for Ubuntu Xenial. Sadly the
steps for Ubuntu Trusty are quite convoluted, created #913 to
improve that. Add the steps for Debian Stretch.

I created a bunch of Dockerfiles to test these instructions.

### CODE REVIEW

**Comment by coryan** [comment]
@frankyn any thoughts?

File: README.md [hunk 1]
```diff
@@ -98,49 +106,97 @@ yum install centos-release-scl
...
 ```
 
-#### Ubuntu (Trusty)
+#### Ubuntu (Trusty Tahr)
+
+The default compiler on Ubuntu-14.04 (Trusty Tahr) has poor support for C++11.
+In addition, gRPC requires a newer version of OpenSSL than the one included
+in the system.
+
+It is possible to work around these limitations on a *fresh* installation of
```
Comment (frankyn) [inline README.md]: s/*fresh*/*new*/

File: README.md [hunk 2]
```diff
@@ -90,6 +94,10 @@ against the latest version of the SDK on each commit and PR.
 
 #### CentOS
 
+The default compiler on CentOS has poor support for C++11. We need to upgrade
```
Comment (frankyn) [inline README.md]: The default compiler on CentOS doesn't fully support C++11.

Comment (coryan) [reply README.md]: Fixed.

File: README.md (hunk 1 above)
Comment (coryan) [reply README.md]: Fixed.

File: README.md [hunk 3]
```diff
@@ -98,49 +106,97 @@ yum install centos-release-scl
...
-sudo update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-6.0 100
-sudo update-alternatives --install /usr/bin/scan-build scan-build /usr/bin/scan-build-6.0 100
-/usr/bin/env GOPATH=/usr go get github.com/bazelbuild/buildtools/buildifier
+sudo apt install -y build-essential cmake git gcc g++ cmake libcurl4-openssl-dev libssl-dev make zlib1g-dev
 ```
 
-#### Ubuntu (Trusty)
+#### Ubuntu (Trusty Tahr)
+
+The default compiler on Ubuntu-14.04 (Trusty Tahr) has poor support for C++11.
```
[the remark this answers is not in the document]
Comment (coryan) [reply README.md]: Fixed.

**Comment by coryan** [comment]
PTAL.

**Comment by frankyn** [comment]
Overall looks much better. The instructions are much more concise. 

Open question is more new Dockerfiles being added to verify README.md are these already included in CI tests?

File: README.md [hunk 4]
```diff
@@ -90,6 +94,10 @@ against the latest version of the SDK on each commit and PR.
 
 #### CentOS
 
+The default compiler on CentOS doesn't fully support C++11. We need to upgrade
+the compiler and other development tools. In this instructions, we use `g++-7`
```
Comment (houglum) [inline README.md]: s/this instructions/these instructions/

File: README.md [hunk 5]
```diff
@@ -188,8 +244,8 @@ cd build-output
 ctest --output-on-failure
 ```
 
-You will find compiled binaries in `build-output\` respective to their source directories.
-
+You will find compiled binaries in `build-output\` respective to their
```
Comment (houglum) [inline README.md]: Should this be `build-output/` (with a forward slash)?

Comment (coryan) [reply README.md]: That is the section for Windows, where it is customary to use `\` for a directory separator, right?

Comment (houglum) [reply README.md]: Bah, GitHub had collapsed the lines between 202 and here -- I thought this was the macOS section.  Nevermind, then :)

File: README.md (hunk 4 above)
Comment (coryan) [reply README.md]: Fixed.

**Comment by coryan** [comment]
@frankyn I forgot to mention that we created #915 as a reminder to automatically check the instructions.

**Comment by frankyn** [comment]
Understood, thanks @coryan, appreciate your patience! LGTM

**Pull request merged**